lib/que/adapters/active_record_with_lock.rb

```diff
@@ -35,13 +47,22 @@ def execute(command, params = [])
       def lock_job_with_lock_database(queue, cursor)
         result = []
         loop do
-          result = Que.execute(:find_job_to_lock, [queue, cursor])
+          result = Que.transaction do
+            observe(nil, FindJobSecondsTotal, queue: queue) do
+              result = Que.execute(:find_job_to_lock, [queue, cursor])
```

Comment:
I don't think it's obvious why we do this, given that there is already an advisory lock. Let's explain in a comment why this improves performance when there are a large number of workers contending for work, and why this doesn't ensure race safety (hence we still need the advisory lock).